### PR TITLE
chore(ts): convert add integration button and child component

### PR DESCRIPTION
**Pull request by scefali**
Fixes: JAVASCRIPT-V3H

### CODE REVIEW

File: src/sentry/static/sentry/app/views/organizationIntegrations/addIntegration.tsx [hunk 1]
```diff
@@ -24,10 +34,11 @@ export default class AddIntegration extends React.Component {
...
   }
 
-  computeCenteredWindow(width, height) {
-    const screenLeft = window.screenLeft !== undefined ? window.screenLeft : screen.left;
+  computeCenteredWindow(width: number, height: number) {
+    const screenLeft =
+      window.screenLeft !== undefined ? window.screenLeft : window.screenX;
 
-    const screenTop = window.screenTop !== undefined ? window.screenTop : screen.top;
+    const screenTop = window.screenTop !== undefined ? window.screenTop : window.screenY;
```
Comment (scefali) [inline src/sentry/static/sentry/app/views/organizationIntegrations/addIntegration.tsx]: TS caught that `screen.top` doesn't exist

File: src/sentry/static/sentry/app/views/organizationIntegrations/addIntegration.tsx [hunk 2]
```diff
@@ -24,10 +33,14 @@ export default class AddIntegration extends React.Component {
     this.dialog && this.dialog.close();
   }
 
-  computeCenteredWindow(width, height) {
-    const screenLeft = window.screenLeft !== undefined ? window.screenLeft : screen.left;
+  dialog: any | null;
```
Comment (scefali) [inline src/sentry/static/sentry/app/views/organizationIntegrations/addIntegration.tsx]: If someone knows a better type for this let me know

File: src/sentry/static/sentry/app/views/organizationIntegrations/installedIntegration.tsx [hunk 3]
```diff
@@ -123,17 +122,16 @@ export default class InstalledIntegration extends React.Component<Props> {
...
-            </Box>
-            <Box>
+            </IntegrationItemBox>
+            <div>
               {integration.status === 'disabled' && (
                 <AddIntegrationButton
                   size="xsmall"
                   priority="success"
                   provider={provider}
-                  integration={integration}
```
Comment (scefali) [inline src/sentry/static/sentry/app/views/organizationIntegrations/installedIntegration.tsx]: This prop isn't on the prop list

File: src/sentry/static/sentry/app/views/organizationIntegrations/installedIntegration.tsx [hunk 4]
```diff
@@ -123,17 +122,16 @@ export default class InstalledIntegration extends React.Component<Props> {
     return (
       <Access access={['org:integrations']}>
         {({hasAccess}) => (
-          <Flex align="center" key={integration.id} className={className}>
-            <Box flex={1}>
+          <IntegrationFlex key={integration.id} className={className}>
+            <IntegrationItemBox>
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/organizationIntegrations/installedIntegration.tsx]: Not sure if intentional, but `flex={1}` translates to `flex: 1`, but I guess having it as a flexbox el itself accomplishes the same thing?

File: src/sentry/static/sentry/app/views/organizationIntegrations/addIntegration.tsx (hunk 2 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/views/organizationIntegrations/addIntegration.tsx]: `Window | null`?

File: src/sentry/static/sentry/app/views/organizationIntegrations/installedIntegration.tsx (hunk 4 above)
Comment (scefali) [reply src/sentry/static/sentry/app/views/organizationIntegrations/installedIntegration.tsx]: @billyvg So I should have:
```
const IntegrationItemBox = styled('div')`
  flex: 1;
`;
```

Comment (billyvg) [reply src/sentry/static/sentry/app/views/organizationIntegrations/installedIntegration.tsx]: Yeah that would be the 1:1 conversion

Comment (scefali) [reply src/sentry/static/sentry/app/views/organizationIntegrations/installedIntegration.tsx]: @billyvg updated

File: src/sentry/static/sentry/app/views/organizationIntegrations/addIntegration.tsx [hunk 5]
```diff
@@ -4,8 +4,17 @@ import queryString from 'query-string';
 
 import {t} from 'app/locale';
 import IndicatorStore from 'app/stores/indicatorStore';
+import {IntegrationProvider, Integration} from 'app/types';
 
-export default class AddIntegration extends React.Component {
+type Props = {
+  children: (openDialog: (urlParams?: any) => void) => React.ReactNode;
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/organizationIntegrations/addIntegration.tsx]: can we make `urlParams` a bit stricter? e.g. `{[key: string}: unknown}`?

File: src/sentry/static/sentry/app/views/organizationIntegrations/addIntegration.tsx [hunk 6]
```diff
@@ -14,8 +23,12 @@ export default class AddIntegration extends React.Component {
     account: PropTypes.string,
   };
 
-  componentDidMount() {
+  constructor(props: Props) {
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/organizationIntegrations/addIntegration.tsx]: we can get rid of this and just declare/init `dialog` as a class property

File: src/sentry/static/sentry/app/views/organizationIntegrations/addIntegration.tsx [hunk 7]
```diff
@@ -24,10 +37,14 @@ export default class AddIntegration extends React.Component {
     this.dialog && this.dialog.close();
   }
 
-  computeCenteredWindow(width, height) {
-    const screenLeft = window.screenLeft !== undefined ? window.screenLeft : screen.left;
+  dialog: Window | null;
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/organizationIntegrations/addIntegration.tsx]: oh we're already doing this, let's initialize it to null and remove constructor

File: src/sentry/static/sentry/app/views/organizationIntegrations/addIntegration.tsx [hunk 8]
```diff
@@ -66,10 +83,10 @@ export default class AddIntegration extends React.Component {
     const opts = `scrollbars=yes,width=${width},height=${height},top=${top},left=${left}`;
 
     this.dialog = window.open(installUrl, name, opts);
-    this.dialog.focus();
+    this.dialog && this.dialog.focus();
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/organizationIntegrations/addIntegration.tsx]: This has been a source of sentry errors, maybe see if you can find the issues on Sentry and link their short IDs to the PR

**Review by billyvg — APPROVED** [review]
Looks good, fixes some bugs too which is 👍 👍 👍 👍

File: src/sentry/static/sentry/app/views/organizationIntegrations/addIntegration.tsx (hunk 8 above)
Comment (scefali) [reply src/sentry/static/sentry/app/views/organizationIntegrations/addIntegration.tsx]: Like this?

Screen Shot 2019-11-22 at 2 06 43 PM

Comment (billyvg) [reply src/sentry/static/sentry/app/views/organizationIntegrations/addIntegration.tsx]: No, it's this 
image

**Pull request merged**